### PR TITLE
charset: add utf8_unicode_ci and utf8mb4_unicode_ci interface

### PR DESCRIPTION
**Pull request by xiongjiwei**
### What problem does this PR solve?

This is first PR of #17596 

### What is changed and how it works?

This PR is aim to add `utf8_unicode_ci` and `utf8mb4_unicode_ci` interface but not implement.

Tests 

- Unit test
- Integration test
### Release note 

- add `utf8_unicode_ci` and `utf8mb4_unicode_ci` interface.

### CODE REVIEW

File: util/collate/unicode_ci.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+func (uc *unicodeCICollator) Key(str string) []byte {
+	return []byte{}
+}
+
+// Pattern implements Collator interface.
+func (uc *unicodeCICollator) Pattern() WildcardPattern {
+	return &unicodePattern{}
+}
+
+// unicodePattern implement use binaryPattern temporary
```
Comment (wjhuang2016) [inline util/collate/unicode_ci.go:34]: unicodePattern implements use binaryPattern temporary.

File: util/collate/unicode_ci.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collate
+
+type unicodeCICollator struct {
+}
+
+// Compare implement Collator interface.
```
Comment (wjhuang2016) [inline util/collate/unicode_ci.go:19]: Compare implements Collator interface.

**Comment by xiongjiwei** [comment]
hi, @wjhuang2016 I changed the comments, is this PR too small?

**Comment by wjhuang2016** [comment]
> hi, @wjhuang2016 I changed the comments. BTW, is this PR too small? and should I not just add interface but implement it

I think it's ok here. Implement an optimized algorithm may be complex and need some benchmarks.

File: util/collate/collate_test.go [hunk 3]
```diff
@@ -164,6 +164,8 @@ func (s *testCollateSuite) TestGetCollator(c *C) {
 	c.Assert(GetCollator("utf8_bin"), FitsTypeOf, &binPaddingCollator{})
 	c.Assert(GetCollator("utf8mb4_general_ci"), FitsTypeOf, &generalCICollator{})
 	c.Assert(GetCollator("utf8_general_ci"), FitsTypeOf, &generalCICollator{})
+	c.Assert(GetCollator("utf8mb4_unicode_ci"), FitsTypeOf, &unicodeCICollator{})
+	c.Assert(GetCollator("utf8_unicode_ci"), FitsTypeOf, &unicodeCICollator{})
```
Comment (wjhuang2016) [inline util/collate/collate_test.go:168]: Please add test cases for GetCollatorByID()

**Comment by xiongjiwei** [comment]
@wjhuang2016 PTAL

File: util/collate/unicode_ci.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collate
+
+type unicodeCICollator struct {
+}
+
+// Compare implement Collator interface. Always return 0 temporary, will change when implement
```
Comment (djshow832) [inline util/collate/unicode_ci.go:19]:
```suggestion
// Compare implements Collator interface. Always return 0 temporary, will change when implement
```

Comment (xiongjiwei) [reply util/collate/unicode_ci.go]: thank you

**Comment by xiongjiwei** [comment]
hi, @djshow832  need do something

**Pull request merged**